src/mito2/src/manifest/storage.rs

```diff
@@ -80,6 +80,19 @@ pub fn file_version(path: &str) -> ManifestVersion {
     s.parse().unwrap_or_else(|_| panic!("Invalid file: {path}"))
 }
 
+/// Return's the file name from path
+/// Just use for .json and .checkpoint file
+/// ### Panics
+/// Panics if the file path is not a valid delta or checkpoint file.
+#[inline]
+pub fn file_name(path: &str) -> String {
+    let name = path.rsplit('/').next().unwrap_or("").to_string();
+    if !is_checkpoint_file(&name) && !is_delta_file(&name) {
+        panic!("Invalid file: {path}")
+    }
+    name
+}
```

Comment:
Prefer not to panic. You can return an `Option` and ignore that file if it isn't a valid file.